lib/semian/circuit_breaker.rb

```diff
@@ -0,0 +1,131 @@
+module Semian
+  class CircuitBreaker
+    extend Forwardable
+    attr_reader :state
+
+    def_delegators :@resource, :acquire, :destroy, :count, :semid
```

Comment:
This stuff seems specific to semaphores. I don't feel like it belongs in this class. After all we have lots of use-cases for CircuitBreaker that have nothing to do with semaphores.